src/containers/IdeLoader.tsx

```diff
@@ -0,0 +1,190 @@
+      await requestWorkspaces();
+    }
+
+    await this.initWorkspace();
+  }
+
+  public async componentDidUpdate(): Promise<void> {
+    const { allWorkspaces, match: { params } } = this.props;
+    const { currentStep } = this.state;
+    const workspace = allWorkspaces.find(workspace =>
+      workspace.namespace === params.namespace && workspace.devfile.metadata.name === params.workspaceName);
+    if (workspace && WorkspaceStatus[workspace.status] === WorkspaceStatus.ERROR) {
+      this.setState({
+        currentStep,
+        hasError: true,
+      });
+      return;
+    }
+    await this.initWorkspace();
+  }
+
+  private async openIDE(workspaceId: string): Promise<void> {
+    await this.props.requestWorkspace(workspaceId);
+    const workspace = this.props.allWorkspaces.find(workspace =>
+      workspace.id === workspaceId);
+    if (!workspace || !workspace.runtime) {
+      return;
+    }
+    let ideUrl = '';
+    const machines = workspace.runtime.machines || {};
+    for (const machineName of Object.keys(machines)) {
+      const servers = machines[machineName].servers || {};
+      for (const serverId of Object.keys(servers)) {
+        const attributes = (servers[serverId] as any).attributes;
+        if (attributes && attributes['type'] === 'ide') {
+          ideUrl = servers[serverId].url;
+          break;
+        }
+      }
+    }
+    if (!ideUrl) {
+      this.showErrorAlert('Don\'t know what to open, IDE url is not defined.');
+      return;
+    }
+    await timeout();
+    this.setState({ currentStep: 4, ideUrl });
+  }
+
+  private async initWorkspace(): Promise<void> {
+    const { allWorkspaces, match: { params } } = this.props;
+    const { namespace, workspaceName } = this.state;
+
+    if (namespace !== params.namespace || workspaceName !== params.workspaceName) {
+      this.setState({ currentStep: 1, hasError: false, ideUrl: '', namespace, workspaceName });
+      await timeout();
+      return;
+    } else if (this.state.currentStep > 2) {
+      return;
+    }
+    const workspace = allWorkspaces.find(workspace =>
+      workspace.namespace === params.namespace && workspace.devfile.metadata.name === params.workspaceName);
+    if (workspace) {
+      this.setState({ currentStep: this.state.currentStep, workspaceId: workspace.id });
+      if (this.state.currentStep === 2 && WorkspaceStatus[workspace.status] === WorkspaceStatus.RUNNING) {
+        this.setState({ currentStep: 3 });
+        return this.openIDE(workspace.id);
+      }
+    } else {
+      this.showErrorAlert('Failed to find the target workspace.');
+      return;
+    }
+    await timeout();
+    this.setState({ currentStep: 2 });
+    if (WorkspaceStatus[workspace.status] === WorkspaceStatus.STOPPED) {
+      try {
+        await this.props.startWorkspace(`${workspace.id}`);
+      } catch (e) {
+        // this.showErrorAlert(`Workspace ${this.state.workspaceName} failed to start. ${e}`);
```

Comment:
I suppose these commented lines should be removed.

Reply:
I have fixed it.